doc/releases/release-notes-3.6.rst

```diff
@@ -476,6 +476,16 @@ Drivers and Sensors
   * New driver :dtcompatible:`analog-axis`.
   * Added ESP32 touch sensor driver including a :dtcompatible:`espressif,esp32-touch`.
 
+* MDIO
+
+  * Fixed initialization priorities of NXP s32 NETC drivers.
+  * Fixed SAM GMAC transfer timeout errors caused by mdio clock not being initialized.
```

Comment:
I would recommend using consistent uppercase spelling of "MDIO" across the entire changeset